source/Classes/004.ManagedIdentityToken.ps1

```diff
@@ -0,0 +1,120 @@
+
+
+class ManagedIdentityToken {
+
+    [SecureString]$access_token
+    [DateTime]$expires_on
+    [int]$expires_in
+    [string]$resource
+    [string]$token_type
+    hidden [bool]$linux = $IsLinux
+
+    # Constructor
+    ManagedIdentityToken([PSCustomObject]$ManagedIdentityTokenObj) {
+
+        # Validate that ManagedIdentityTokenObj is a HashTable and Contains the correct keys
+        if (-not $this.isValid($ManagedIdentityTokenObj)) { throw "The ManagedIdentityTokenObj is not valid." }
+
+        $epochStart = [datetime]::new(1970, 1, 1, 0, 0, 0, [DateTimeKind]::Utc)
+
+        # Set the properties of the class
+        $this.access_token  = $ManagedIdentityTokenObj.access_token | ConvertTo-SecureString -AsPlainText -Force
+        $this.expires_on    = $epochStart.AddSeconds($ManagedIdentityTokenObj.expires_on)
+        $this.expires_in    = $ManagedIdentityTokenObj.expires_in
+        $this.resource      = $ManagedIdentityTokenObj.resource
+        $this.token_type    = $ManagedIdentityTokenObj.token_type
+
+    }
+
+    # Function to validate the ManagedIdentityTokenObj
+    hidden [bool]isValid($ManagedIdentityTokenObj) {
+
+        # Assuming these are the keys we expect in the hashtable
+        $expectedKeys = @('access_token', 'expires_on', 'expires_in', 'resource', 'token_type')
+
+        # Check if all expected keys exist in the hashtable
+        foreach ($key in $expectedKeys) {
+            if (-not $ManagedIdentityTokenObj."$key")
+            {
+                Write-Verbose "[ManagedIdentityToken] The hashtable does not contain the expected property: $key"
+                return $false
+            }
+        }
+
+        # If all checks pass, return true
+        Write-Verbose "[ManagedIdentityToken] The hashtable is valid and contains all the expected keys."
+        return $true
+    }
+
+    # Function to convert a SecureString to a String
+    hidden [string]ConvertFromSecureString([SecureString]$SecureString) {
+        # Convert a SecureString to a String
+        $BSTR = [System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($SecureString)
+        $String = ($this.linux) ? [System.Runtime.InteropServices.Marshal]::PtrToStringUni($BSTR) :
```

Comment:
based on discussion can we change this to be able to back to v5 syntax and check based on discussion above